cmd/kraft/run/run.go

```diff
@@ -303,10 +306,26 @@ func (opts *Run) Run(cmd *cobra.Command, args []string) error {
 			machine.Status.StateDir = filepath.Join(config.G[config.KraftKit](ctx).RuntimeDir, string(machine.ObjectMeta.UID))
 		}
 
-		if err := os.MkdirAll(machine.Status.StateDir, 0o755); err != nil {
+		if err := os.MkdirAll(machine.Status.StateDir, fs.ModeSetgid|0o775); err != nil {
 			return fmt.Errorf("could not make machine state dir: %w", err)
 		}
 
+		group, err := user.LookupGroup("kraftkit")
```

Comment:
I'd turn this into a config option,
```go
group, err := user.LookupGroup(config.G[config.KraftKit](ctx).UserGroup)
```

And set the default to `kraftkit` in the new entry.